src/context/CategoriesContext/index.tsx

```diff
@@ -0,0 +1,74 @@
+import { createContext, ReactNode, useState } from 'react';
+
+import { MOCKED_CATEGORY_LIST, MOCKED_CLASSES, MOCKED_SAVED_COURSES } from '../../../.mocks/constants'
+import { ICategory, ICourse, IClasses } from '@models'
+
+interface ICategoriesContextProps {
+  allCategories: ICategory[];
+  savedCategories: ICategory[];
+  currentClasses: ICourse[] | undefined;
+  getCategory: (id: string) => ICategory | undefined;
+  getCategoryClasses: (id: string) => ICourse[] | undefined;
+  addCategory: (id: string) => ICategory[];
+  deleteCategory: (id: string) => ICategory[];
+}
+
+interface ICategoriesProviderProps {
+  children: ReactNode;
+}
+
+export const CategoriesContext = createContext({} as ICategoriesContextProps);
+
+export function CategoriesProvider({ children }: ICategoriesProviderProps) {
+
+  const [allCategories] = useState<ICategory[]>(MOCKED_CATEGORY_LIST);
+  const [allCategoriesClasses] = useState<IClasses[]>(MOCKED_CLASSES);
+  const [savedCategories, setSavedCategories] = useState<ICategory[]>(MOCKED_SAVED_COURSES);
+  const [currentClasses, setCurrentClasses] = useState<ICourse[] | undefined>([]);
```

Comment:
Acho que o current que falamos da outra vez se aplica aqui também. 
Outro ponto é que não sei se faz sentido o classes aqui ser undefined, o mais interessante seria você tratar que ele sempre vai ter um valor, mesmo que esse valor seja um array vazio.

Reply:
resolvido a questão dos nomes.. agora essa questão do undefined, não to sabendo resolver.. o typescript reclama dizendo que pode ser undefined aqui:
![image](https://user-images.githubusercontent.com/67794156/172912038-ebb11703-f10d-49e9-87ab-d902040b5c7d.png)

Reply:
Então, ai já é o find que pode te devolver undefined caso a categoria não seja encontrada na lista, neste caso, faz sentido chamar o setClasses quando você não encontra a categoria? Ou você deveria mostrar uma mensagem de erro?

Reply:
teoricamente isso nunca aconteceria, pq o jeito de chegar nessa rota é passando pelo category.. ha nao ser se a pessoa tivesse em mãos uma url inválida. Mas acho válido tratar esse caso sim. Na última MR (testes de hooks) tem um refactor desse cara.. veja o que acha

Reply:
Beleza, segue o baile com esse e se for o caso a gente vê depois.